ad_block_client.cc

```diff
@@ -470,6 +470,7 @@ AdBlockClient::AdBlockClient() : filters(nullptr),
   numHashSetSaves(0),
   numExceptionHashSetSaves(0),
   deserializedBuffer(nullptr) {
+  g_unknownOptions = new std::set<std::string>();
```

Comment:
We can't create a global member here because there can be multiple adblock clients in use.  browser-laptop has several for example. 
So it would lead to a memory leak there.  
For example we even create one every time a user makes a custom rule to test it. 
This would make all adblock clients unusable after the user typed any rule.

Reply:
Let's just make it an std::unique_ptr in adblockclient's header itself.